openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -112,18 +162,60 @@ def _check_for_pacing_changes(previous_course_overview, updated_course_overview)
+    """
+    Checks if the certificate available date (CAD) or the certificates display behavior (CDB) of a course run has
+    changed during a course overview update. If so, we emit the COURSE_CERT_DATE_CHANGE signal to ensure other parts of
+    the system are aware of the change. The `credentials` app listens for this signal in order to keep our certificate
+    visibility settings in the Credentials IDA up to date.
+
+    Args:
+        previous_course_overview (CourseOverview): the current course overview instance for a particular course run
+        updated_course_overview (CourseOverview): an updated course overview instance, reflecting the current state of
+            data from the modulestore/Mongo
+
+    Returns:
+        None
+    """
+    def _send_course_cert_date_change_signal():
+        """
+        A callback used to fire the COURSE_CERT_DATE_CHANGE Django signal *after* the ORM has successfully commit the
+        update.
+        """
+        COURSE_CERT_DATE_CHANGE.send_robust(sender=None, course_key=str(updated_course_overview.id))
+
+    course_run_id = str(updated_course_overview.id)
+    prev_available_date = previous_course_overview.certificate_available_date
+    prev_display_behavior = previous_course_overview.certificates_display_behavior
+    prev_end_date = previous_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+    updated_available_date = updated_course_overview.certificate_available_date
+    updated_display_behavior = updated_course_overview.certificates_display_behavior
+    updated_end_date = updated_course_overview.end  # `end_date` is a deprecated field, use `end` instead
+
+    send_signal = False
+    if prev_available_date != updated_available_date:
         LOG.info(
-            f"Certificate availability date for {str(updated_course_overview.id)} has changed from " +
-            f"{previous_course_overview.certificate_available_date} to " +
-            f"{updated_course_overview.certificate_available_date}. Sending COURSE_CERT_DATE_CHANGE signal."
+            f"The certificate available date for {course_run_id} has changed from {prev_available_date} to "
+            f"{updated_available_date}. Firing COURSE_CERT_DATE_CHANGE signal."
         )
+        send_signal = True
+    elif prev_display_behavior != updated_display_behavior:
+        LOG.info(
+            f"The certificates display behavior for {course_run_id} has changed from {prev_display_behavior} to "
+            f"{updated_display_behavior}. Firing COURSE_CERT_DATE_CHANGE signal."
+        )
+        send_signal = True
+    # edge case -- if a course run with a cert display behavior of "End date of course" has changed its end date, we
+    # should fire our signal to ensure visibility of certificates managed by the Credentials IDA are corrected too
+    elif (
+        (prev_display_behavior == CertificatesDisplayBehaviors.END and
+         updated_display_behavior == CertificatesDisplayBehaviors.END) and
```

Comment:
if these conditions stay as `elif`, then this is unnecessarily complex; you've already  short-circuited out on `!=`  on the previous conditional.

 also, does it matter if they are _both_ set to `CertificatesDisplayBehaviors.END`? You are already sending a signal if they differ from each other, so if they've changed display behavior you are already sending an update signal. I would think that all you need to know is 

```
if  updated_display_behavior == CertificatesDisplayBehaviors.END and prev_end_date != updated_end_date
```

right?

Reply:
Yep, I think you're correct. Having unnecessarily complex code is what got me in this situation in the first place 😆. Thanks for the attention here.